test/download-test.js

```diff
@@ -273,6 +274,28 @@ test('error during download should fail with no dangling temp file', function (t
   })
 })
 
+test('should fail if abi is system abi with invalid binary', function (t) {
+  var opts = getOpts()
+  opts.abi = rc.abi
+  opts.pkg.binary = {host: 'http://localhost:8890'}
+
+  var server = http.createServer(function (req, res) {
+    res.statusCode = 200
+    var archive = path.join(__dirname, 'invalid.tar.gz')
```

Comment:
`invalid.tar.gz` just contains a text file renamed to `invalid.node` that should fail when it's required